libs/dal/src/repositories/topic/topic.repository.ts

```diff
@@ -0,0 +1,25 @@
+import { AuthProviderEnum } from '@novu/shared';
+import { FilterQuery } from 'mongoose';
+
+import { EnvironmentId, OrganizationId, TopicKey, TopicEntity } from './topic.entity';
+import { Topic } from './topic.schema';
+
+import { BaseRepository } from '../base-repository';
+
+export class TopicRepository extends BaseRepository<FilterQuery<TopicEntity>, TopicEntity> {
+  constructor() {
+    super(Topic, TopicEntity);
+  }
+
+  async findTopicByKeyOrganizationAndEnvironment(
```

Comment:
Same comment from my previous PR goes here. 
```suggestion
  async findTopicByKey(
```